TACHO_DRIVER.md

```diff
@@ -0,0 +1,199 @@
+|   100               | Motor ramps up to 100% in 1000 msec
+|   50                | Motor ramps down to 50% in 1000 msec
+|   -50               | Motor ramps down to 0% in 1000 msec and up to -50 in 2000 msec
+
+## Run Modes
+
+The `run` attribute determines function that the motor is executing. Within a `run`
+mode, some attributes have an effect, while others do not. These details are
+described below.
+
+The run modes can be classified into two types. The `run-direct` is suited to
+applications where the programmer wants direct control of the motors. The other
+three modes:
+
+- `run_timed`
+- `run_to_abs_pos`
+- `run_to_rel_pos`
+
+They can be through of as convenience commands where the driver handles the motor
+control independently of the application. This is well suited to applications where you
+want the driver to handle the details, like machines where the motors must alternate
+between precise positions or run for fixed timing periods.
+
+### `run-direct` Mode
+
+In its most basic operating mode, the `tacho-motor` accepts the `run-direct`
+command. After accepting a `run-direct` command, the motor will only accept
+a `stop` command - the other commands represent "helpers" that automatically
+do what `run-direct` is capable of without user intervention.
+
+NOTE: In this mode the `stop` command stops the motor immediately! If you
+want the motor to use the ramping feature, change the desired setpoint to 0.
+
+Currently, only changes to the `duty_cycle_sp` take effect
+immediately and I'm proposing that this should depend on the `speed_regulation`
+setting. I may be convinced to also allow changes to the ramp setpoints.
+
+You cannot change the `speed_regulation` value without stopping the
+motor first. (Not sure on this - I can be convinced to change this)
+
+Finally, if the ramping target setpoint has been set to 0, the motor is
+stopped in the corresponding `stop_mode`.
+
+### `run-forever` Mode
+
+Deprecated? This does not provide any new features that are not already present
+in `run-direct`
+
+### `run-timed`, `run_to_abs_pos` and `run_to_rel_pos` Modes
+
+These are the convenience commands described earlier. The motors will not
+respond to changes in any setpoints or command except for `stop`. The `stop`
+command turns the motor off immediately in whatever the `stop_mode` is set
+to - no ramping will take place.
+
+I may be convinced to make the `stop` command leave the motor in `float` mode
+for safety.
+
+## Stall Detection
+
+Stall detection is something that can be complicated, and we don't want to
+report false positives when stalled. In its simplest form, the motor can
+be considered stalled when there has been less than `x` pulses of change
+in the last `y` units of time when the absolute value of the duty cycle is
+greater than `z`.
+
+So we probably need three more attributes to describe the stall condition
+and another to report the stall state. And we'll need to handle the case
+of the motor in the hold state.
```

Comment:
These are already implemented as flags of the `state` attribute. They work for PiStorms and NxtMMX, just not implemented on EV3 yet.